elastic/kibana
# [Stack Monitoring] Fix hashchange detection on sidenav link

## Summary

Fixes #112431 

Uses established technique from the Graph plugin (https://github.com/elastic/kibana/blob/master/x-pack/plugins/graph/public/application.ts#L108-L118) to broadcast hashchanges from the global Kibana history object so that React-Router can pick up on them. This allows clicking the sidenav link from within Stack Monitoring to work again.

### To test

1. Set `monitoring.ui.render_react_app: true` in `kibana.dev.yml`
2. Open the Stack Monitoring app, and navigate away from the overview page
3. Open the sidenav, and click Stack Monitoring again
4. Ensure that the url changes to `/app/monitoring` with no hash
5. Ensure that the URL immediately redirects to `#/loading` and then to `#/overview`, and that the Overview page loads
